drivers/stepper/gpio_stepper_controller.c

```diff
@@ -53,20 +55,35 @@ static int stepper_motor_set_coil_charge(const struct device *dev)
 	return 0;
 }
 
+static void increment_coil_charge(const struct device *dev)
+{
+	struct gpio_stepper_data *data = dev->data;
+
+	data->coil_charge =
+		data->coil_charge == NUM_CONTROL_PINS * MAX_MICRO_STEP_RES - data->step_gap
+			? 0
+			: data->coil_charge + data->step_gap;
```

Comment:
yikes, can you expand it to a normal `if`? the indentation is something else here